server/lib/src/idm/account.rs

```diff
@@ -683,6 +704,38 @@ impl Account {
             }
         }
     }
+
+    pub fn verify_application_password(
+        &self,
+        app_uuid: &Uuid,
+        app_label: &str,
+        cleartext: &str,
+        ct: Duration,
+    ) -> Result<Option<UserAuthToken>, OperationError> {
+        for ap in self.apps_pwds.values() {
```

Comment:
Why iterate here? You can dho self.apps_pwds.get(app_uuid) to get all the passwords related to that application uuid, and then you limit how many iterations you need to do.